maplibre/src/stages/mod.rs

```diff
@@ -1,187 +1,19 @@
-mod request_stage;
 
 /// Register stages required for requesting and preparing new tiles.
 pub fn register_stages<HC: HttpClient, SM: ScheduleMethod>(
     schedule: &mut Schedule,
     http_source_client: HttpSourceClient<HC>,
     scheduler: Box<Scheduler<SM>>,
 ) {
-    let (message_sender, message_receiver): (MessageSender, MessageReceiver) = mpsc::channel();
-    let shared_thread_state = SharedThreadState {
-        tile_request_state: Arc::new(Mutex::new(TileRequestState::new())),
-        message_sender,
-        geometry_index: Arc::new(Mutex::new(GeometryIndex::new())),
-    };
-
-    schedule.add_stage(
-        "request",
-        RequestStage::new(shared_thread_state.clone(), http_source_client, *scheduler),
-    );
-    schedule.add_stage(
-        "populate_tile_store",
-        PopulateTileStore::new(shared_thread_state, message_receiver),
-    );
-}
-
-pub struct HeadedPipelineProcessor {
-    state: SharedThreadState,
-}
-
-impl PipelineProcessor for HeadedPipelineProcessor {
-    fn tile_finished(&mut self, request_id: TileRequestID, coords: &WorldTileCoords) {
-        self.state
-            .message_sender
-            .send(TessellateMessage::Tile(TileTessellateMessage {
-                request_id,
-                coords: *coords,
-            }))
-            .unwrap();
-    }
-
-    fn layer_unavailable(&mut self, coords: &WorldTileCoords, layer_name: &str) {
-        self.state
-            .message_sender
-            .send(TessellateMessage::Layer(
-                LayerTessellateMessage::UnavailableLayer {
-                    coords: *coords,
-                    layer_name: layer_name.to_owned(),
-                },
-            ))
-            .unwrap();
-    }
-
-    fn layer_tesselation_finished(
-        &mut self,
-        coords: &WorldTileCoords,
-        buffer: OverAlignedVertexBuffer<ShaderVertex, IndexDataType>,
-        feature_indices: Vec<u32>,
-        layer_data: tile::Layer,
-    ) {
-        self.state
-            .message_sender
-            .send(TessellateMessage::Layer(
-                LayerTessellateMessage::TessellatedLayer {
-                    coords: *coords,
-                    buffer,
-                    feature_indices,
-                    layer_data,
-                },
-            ))
-            .unwrap();
-    }
-
-    fn layer_indexing_finished(
-        &mut self,
-        coords: &WorldTileCoords,
-        geometries: Vec<IndexedGeometry<f64>>,
-    ) {
-        if let Ok(mut geometry_index) = self.state.geometry_index.lock() {
-            geometry_index.index_tile(coords, TileIndex::Linear { list: geometries })
-        }
-    }
-}
-
-/// Stores and provides access to the thread safe data shared between the schedulers.
-#[derive(Clone)]
-pub struct SharedThreadState {
-    pub tile_request_state: Arc<Mutex<TileRequestState>>,
-    pub message_sender: mpsc::Sender<TessellateMessage>,
-    pub geometry_index: Arc<Mutex<GeometryIndex>>,
-}
```

Comment:
:fire:  We won't be be able to use Mutex on the web in the future because of COOP! So this is a very good change!